src/libraries/Microsoft.PowerFx.Core/Binding/BindingConfig.cs

```diff
@@ -19,13 +19,16 @@ internal class BindingConfig
 
         public bool UseThisRecordForRuleScope { get; }
 
-        public bool NumberIsFloat { get; }
+        public bool NumberIsFloat { get; }
+
+        public bool AnalysisMode { get; }
```

Comment:
Probably would be better as an enum that "self documents" a bit better. 

Maybe { InferredTypesAllowed, DeclaredTypesOnly } or something like that.